### DISCUSSION TITLE
Worker not sending mails ,need to retart the postal again and again.

### DISCUSSION THREAD

**Discussion by Shamri04** [postalserver/postal#2300] [Help with using Postal]
Hi 

I am having postal with 2.1.2 installed with a docker container, Mails in the queue are not delivering. I have seen the output of the SQL queue but looks no lock there. I think the worker is getting stuck and not processing the emails. kindly help with this

**Comment by willpower232** [comment]
Can you share the output of `postal status`?

**Comment by Shamri04** [comment]
It is often showing all the services are running. As checked there is some error with the rabbitmq
[
Attaching the logs
9] [2022-12-21T04:16:39.032] INFO -- : [W2OMO95KNO] [1::586513 16765] /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/consumer_work_pool.rb:102:in `run_loop'
[9] [2022-12-21T04:16:39.035] INFO -- : [W2OMO95KNO] [1::586513 16765] Queued message was unlocked
[9] [2022-12-21T04:16:39.040] INFO -- : [RPRLMACK] Finishing up
[9] [2022-12-21T04:16:39.349] INFO -- : [W2OMO95KNO] Finished processing UnqueueMessageJob job in 1825.315217943s
E, [2022-12-21T04:16:39.350254 #9] ERROR -- #<Bunny::Session:0x557fa2be6d70 postal@172.24.131.131:5672, vhost=postal, addresses=[172.24.131.131:5672]>: Uncaught exception from consumer #<Bunny::Consumer:47003340013040 @channel_id=1 @queue=deliver-jobs-main> @consumer_tag=bunny-1671590473000-882226687407>: #<Bunny::ChannelAlreadyClosed: cannot use a closed channel! Channel id: 1, closed due to a server-reported channel error: PRECONDITION_FAILED - delivery acknowledgement on channel 1 timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more> @ /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/channel.rb:1999:in `raise_if_no_longer_open!'

**Reply by willpower232** [reply]
Is your rabbitmq server running and does it have enough memory?

**Comment by masihfathi** [comment]
@Shamri04 did you found any solution?
